### PR TITLE
Disable StoreAPI if agent enabled

### PR DESCRIPTION
**Pull request by Vanshikav123**
## Changes
Made changes in storeset.go by adding a extra variable which checks whether the agent is enabled or not.
#6606

### CODE REVIEW

**Comment by Vanshikav123** [comment]
hello @yeya24 I am trying to implement this can you please suggest your feedback regarding this error and the changes that i have done.

File: pkg/query/internal/test-storeset-pre-v0.8.0/storeset_test.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func TestPre0_8_0_StoreSet_AgainstNewStoreGW(t *testing.T) {
 	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
 	logger = level.NewFilter(logger, level.AllowDebug())
 	logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
-	storeSet := NewStoreSet(logger, nil, specsFromAddrFunc(st.StoreAddresses()), testGRPCOpts, time.Minute)
+	storeSet := NewStoreSet(logger, nil, specsFromAddrFunc(st.StoreAddresses(), IsAgentEnabled), testGRPCOpts, time.Minute)
```
Comment (yeya24) [inline pkg/query/internal/test-storeset-pre-v0.8.0/storeset_test.go:189]: As the error suggested, the variable `IsAgentEnabled` is not defined. Sidecar needs to automatically detect it based on the result of the flags API.

**Comment by yeya24** [comment]
@Vanshikav123 Please make your changes can be compiled first.

File: pkg/query/internal/test-storeset-pre-v0.8.0/storeset_test.go (hunk 1 above)
Comment (Vanshikav123) [reply pkg/query/internal/test-storeset-pre-v0.8.0/storeset_test.go:189]: Thank you @yeya24 I think I headed to find the solution for this issue in wrong direction sorry for the inconvenience I will Close this PR and open a new PR and work on this issue to fix it as soon as possible.

Comment (Vanshikav123) [reply pkg/query/internal/test-storeset-pre-v0.8.0/storeset_test.go:189]: Thank you @yeya24 I think I headed to find the solution for this issue in wrong direction sorry for the inconvenience I will work on this to fix it as soon as possible.

**Pull request closed without merging**